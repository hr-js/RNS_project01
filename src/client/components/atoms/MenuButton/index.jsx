// @flow
import * as React from "react";
import styles from "./style.css";

type PropsType = {
  children: string,
  selected?: string,
  onButtonClick: () => void
};

function MenuButton(props: PropsType): React.Node {
  const className: string =
    props.selected === "selected"
      ? `${styles.button} ${styles.selected}`
      : styles.button;
  return (
    <button className={className} onClick={props.onButtonClick}>
      {props.children}
    </button>
  );
}
export default MenuButton;
