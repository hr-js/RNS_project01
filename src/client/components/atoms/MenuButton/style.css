@charset "utf-8";
.button {
  width: 100%;
  height: 85px;
  margin: 0;
  padding: 0;
  font-size: 18px;
  color: gray;
  background: transparent;
  border: none;
  border-bottom: 1px dashed silver;
  box-sizing: border-box;
  transition: font-size 200ms;
  outline: none;
}
.button.selected {
  background-color: silver;
  border: none;
  color: white;
  font-size: 24px;
  font-weight: bold;
}
