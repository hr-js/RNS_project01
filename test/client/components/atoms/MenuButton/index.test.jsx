import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import MenuButton from "@components/atoms/MenuButton";

describe("MenuButton.jsxのテスト", () => {
  const func = () => {};
  describe("スナップショット", () => {
    it("renders corecctly", () => {
      const tree = renderer
        .create(
          <MenuButton selected="selected" onButtonClick={func}>
            メニュー
          </MenuButton>
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("コンポーネントのテスト", () => {
    describe("子要素に「入退室処理」を指定して、ボタンをクリックした時", () => {
      const onButtonClick = jest.fn();
      const button = shallow(
        <MenuButton onButtonClick={onButtonClick}>入退室処理</MenuButton>
      );
      button.find("button").simulate("click");
      it("子要素は「入退室処理」である", () => {
        expect(button.text()).toBe("入退室処理");
      });
      it("onButtonClickがコールされた", () => {
        expect(onButtonClick).toHaveBeenCalled();
      });
    });
    describe("selected属性を使用しない場合", () => {
      const button = shallow(
        <MenuButton onButtonClick={() => {}}>入退室処理</MenuButton>
      );
      it("classNameは「button」である", () => {
        expect(button.props().className).toBe("button");
      });
    });
    describe("selected属性を使用した場合", () => {
      const button = shallow(
        <MenuButton selected="selected" onButtonClick={() => {}}>
          入退室処理
        </MenuButton>
      );
      it("classNameは「button selected」である", () => {
        expect(button.props().className).toBe("button selected");
      });
    });
  });
});
