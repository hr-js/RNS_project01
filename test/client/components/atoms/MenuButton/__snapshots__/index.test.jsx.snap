// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuButton.jsxのテスト スナップショット renders corecctly 1`] = `
<button
  className="button selected"
  onClick={[Function]}
>
  メニュー
</button>
`;
